### ISSUE TITLE
DOMContentLoaded is missing from autocomlete

### ISSUE THREAD

**Issue by TheColorRed** [microsoft/TypeScript#20867]
**TypeScript Version:**  2.6.2

**Code**

```ts
document.addEventListener('DOMContentLoaded', () => { /* Do something */ })
```

**Expected behavior:**

When typing `DOMContentLoaded` into the string, it should be suggested in the intellisense

**Actual behavior:**

All other auto completion items are suggested (click, drag, drop, etc.) except for `DOMContentLoaded`

**Comment by TheColorRed** [comment]
I think it just needs to be added to the definition file:

https://github.com/Microsoft/TypeScript/blob/master/lib/lib.d.ts#L6644

**Comment by DanielRosenwasser** [comment]
I'm not sure how easy it will be to change this given that https://github.com/Microsoft/TSJS-lib-generator/blob/a174e93e4bc20dde392afd3e94a1d5bf526b0bf0/inputfiles/browser.webidl.xml#L3876 is generated.

**Comment by svipben** [comment]
Any news about this? Maybe we can help somehow to include `DOMContentLoaded`?

**Comment by erezbm** [comment]
Looks like the solution is adding `"DOMContentLoaded": Event;` to `DocumentEventMap` ([here](https://github.com/microsoft/TypeScript/blob/0ba250dc3bf0daface7b861c5ccb6b4b9baf78ac/lib/lib.dom.d.ts#L4405)) and `WindowEventMap` ([here](https://github.com/microsoft/TypeScript/blob/0ba250dc3bf0daface7b861c5ccb6b4b9baf78ac/lib/lib.dom.d.ts#L18348)).

**Comment by MasterGordon** [comment]
> Looks like the solution is adding `"DOMContentLoaded": Event;` to `DocumentEventMap` ([here](https://github.com/microsoft/TypeScript/blob/0ba250dc3bf0daface7b861c5ccb6b4b9baf78ac/lib/lib.dom.d.ts#L4405)) and `WindowEventMap` ([here](https://github.com/microsoft/TypeScript/blob/0ba250dc3bf0daface7b861c5ccb6b4b9baf78ac/lib/lib.dom.d.ts#L18348)).

The files you have linked are generated with this Tool: https://github.com/microsoft/TypeScript-DOM-lib-generator

**Comment by khokm** [comment]
I simply dont understand... Why is it so hard for developers to simply add ONE SINGLE STRING to generator, so we dont need to fix it it by yourself every time after updating TS version?

**Comment by LachlanArthur** [comment]
This is now available in TypeScript 4.7 (released yesterday)

![image](https://user-images.githubusercontent.com/1870204/170384618-86a0a51c-1b12-47a8-b80a-e89d86984b7a.png)

**Comment by TheColorRed** [comment]
Closing as this is now implemented.
